conans/client/toolchain/cmake/__init__.py

```diff
@@ -0,0 +1,16 @@
+from conans.client.tools import cross_building
+from .native import CMakeNativeToolchain
+from .android import CMakeAndroidToolchain
+
+
+def CMakeToolchain(conanfile, **kwargs):
+    if not cross_building(conanfile):
+        return CMakeNativeToolchain(conanfile=conanfile, **kwargs)
+    else:
+        # Exceptions to cross-building scenarios
+        if conanfile.settings.os == 'Windows' and conanfile.settings_build.os == 'Windows':
```

Comment:
why exceptions for Windows? anyway, this sounds like out of scope for Android, don't understand why it should be changed here

Reply:
More about it here (https://github.com/conan-io/conan/pull/7841), this PR is built on top of that one. But, specifically, comments related to the Windows exception here (reviews and comments: https://github.com/conan-io/conan/pull/7827), current Conan behavior doesn't consider a cross-building scenario when going from x64 to x86 in Windows or Linux.